containers/podman-desktop
# feat: improved empty screens

### What does this PR do?

Improves the empty-screen message when the cause is a filter that doesn't match anything.

The design requires adding second line in EmptyScreen, which I've done as a detail message. I then looked at making a *FilterEmptyScreen for each page as it's own component or inline, but there was a lot of copy/paste so I created a generic FilteredEmptyScreen. As per the design in issue #3602, when the filter is too long (I picked 20 chars) the search term isn't shown.

searchTerm was already exported from ContainerList, I exported it from the others as well since it is necessary to test, and a useful feature in general. Tests added to show the filter empty screen comes up when the filter doesn't match anything.

### Screenshot/screencast of this PR

https://github.com/containers/podman-desktop/assets/19958075/13e724f7-252f-4915-bb04-00665001d21d

### What issues does this PR fix or reference?

Fixes #3602.

### How to test this PR?

Type a filter that matches (to confirm no regression), doesn't match (to confirm new behaviour), and is really long (to show the message changes).
